src/portable/st/stm32_fsdev/dcd_stm32_fsdev_pvt_st.h

```diff
@@ -296,6 +296,19 @@ TU_ATTR_ALWAYS_INLINE static inline uint32_t pcd_get_ep_rx_cnt(USB_TypeDef * USB
 {
 #ifdef FSDEV_BUS_32BIT
   (void) USBx;
+  volatile uint32_t count = 10;
+  /*
+  WA: few cycles for RX PMA descriptor to update
+  This workaround is ported from stm32h5xx_hal_pcd.h : PCD_GET_EP_RX_CNT H5
+  This code fixes an issue when the code is compiled in GCC with a fast optimization(O2/O3) and device with an high frequency.
+  The function doesn't return the correct value.
+  Issue observed on Windows 10 and stm32h573i_dk and tud_task() scheduled by IT, the device is not migrated the USB device is not visible .
+  */
+  while (count > 0U)
+  {
+    asm("NOP");
+    count--;
```

Comment:
Original solution from ST does not need NOP, since count-- does also take 1 cycle. I think the goal is wait 10 cycles, do you have issue if we drop NOP instruction here ?
```C
  /* WA: few cycles for RX PMA descriptor to update */
  while (count > 0U)
  {
    count--;
  }
```

Reply:
@Lurcy38 could you please confirm if we could remove the 'nop' instruction and just keep the count-- is enough ?

Reply:
yes, you can remove the "nop", I added this nop by safty, to consume more time in case of higher frequency and force the compiler to keep the loop as is

Reply:
thank I remove the nop instruction, will merge when ci all passsed.